### PR TITLE
Implement Quorum approval

### PR DESCRIPTION
**Pull request by zugzwang**
It would be nice to be able to perform ECDH with inline EC public keys instead of relying on transient keys for ECDH exchanges.

### CODE REVIEW

**Comment by zugzwang** [comment]
Fixes #57

**Comment by zugzwang** [comment]
This implements quorum approval. `sq-dsm` loops waiting until the approval is accepted. (Run with `RUST_LOG=info`)

**Comment by zugzwang** [comment]
Implement 2-step instead of waiting.

File: openpgp-dsm/src/lib.rs
Comment (MihirLuthra) [inline openpgp-dsm/src/lib.rs:491]: nit: can we somehow add a prefix for dsm specific logs to make filtering out easier?
like:
```rust
macro_rules! sq_dsm_log {
    ($log_type:path, $($log:tt)*) => {{
        $log_type ! ("sq-dsm log: {}", std::format!($($log)*));
    }};
}

// use like
sq_dsm_log!(info, "generate certificate - self-sign primary key");
```

File: openpgp-dsm/src/lib.rs
Comment (MihirLuthra) [inline openpgp-dsm/src/lib.rs:1046]: I see we use `expect` at some places instead of `context` but they appear to be regular errors too. Can you correct me if I am wrong here?

File: openpgp-dsm/src/lib.rs
Comment (zugzwang) [reply openpgp-dsm/src/lib.rs:132]: This requires the pending approval's UUID, but good idea, let's the user handle it

File: openpgp-dsm/src/lib.rs
Comment (zugzwang) [reply openpgp-dsm/src/lib.rs:127]: Definitely

File: openpgp-dsm/src/lib.rs
Comment (zugzwang) [reply openpgp-dsm/src/lib.rs:132]: On second thought, one key generation requires 4 signatures + 1 update at the moment. The signatures may perhaps be submitted in parallel, but the update (storing PGP metadata at DSM) needs to be done afterwards and remember the signatures, e.g. this is stateful.

File: openpgp-dsm/src/lib.rs
Comment (zugzwang) [reply openpgp-dsm/src/lib.rs]: Removed all expects - trailing from a version without proper error handling. Thanks.

**Review by MihirLuthra — APPROVED** [review]
Looks good to me.

**Pull request merged**
